components/script/dom/htmlscriptelement.rs

```diff
@@ -243,42 +366,85 @@ impl FetchResponseListener for ClassicContext {
 
     /// <https://html.spec.whatwg.org/multipage/#fetch-a-classic-script>
     /// step 4-9
+    #[allow(unsafe_code)]
     fn process_response_eof(&mut self, response: Result<ResourceFetchTiming, NetworkError>) {
-        // Step 5.
-        let load = response.and(self.status.clone()).map(|_| {
-            let metadata = self.metadata.take().unwrap();
+        let (source_text, final_url) = match (response, self.status.as_ref()) {
+            (Err(err), _) => {
+                // Step 6, response is an error.
+                finish_fetching_a_classic_script(
+                    &*self.elem.root(),
+                    self.kind.clone(),
+                    self.url.clone(),
+                    Err(err),
+                );
+                return;
+            },
+            (_, Err(err)) => {
+                // Step 6, response is an error.
+                finish_fetching_a_classic_script(
+                    &*self.elem.root(),
+                    self.kind.clone(),
+                    self.url.clone(),
+                    Err(err.clone()),
+                );
+                return;
+            },
```

Comment:
Could we combine these as `(Err(err), _) | (_, Err(err))` and avoid the duplication?

Reply:
There was an issue because the `_` is not the same type in those two arms(there might be another way to make it work, but that is why I added two different match arms).

Reply:
Are you sure it's the _ that matters? What if you matched on `response.as_ref()`?

Reply:
I'm not sure, I'll leave it to @AbhishekSharma102 to look into.

Reply:
Using `response.as_ref()` doesn't throw any errors. Have merged the two